library/core/src/option.rs

```diff
@@ -90,6 +90,205 @@
+//!
+//! [`&mut Pin`]: crate::pin::Pin
+//! [`&Pin`]: crate::pin::Pin
+//! [`as_deref`]: Option::as_deref
+//! [`as_deref_mut`]: Option::as_deref_mut
+//! [`as_mut`]: Option::as_mut
+//! [`as_pin_ref`]: Option::as_pin_ref
+//! [`as_pin_mut`]: Option::as_pin_mut
+//! [`as_ref`]: Option::as_ref
+//! [`Pin`]: crate::pin::Pin
+//!
+//! ## Extracting the contained value
+//!
+//! These methods extract the contained value in an [`Option`] when it is
+//! the [`Some`] variant. If the [`Option`] is [`None`]:
+//!
+//! * [`expect`] panics with a provided custom message
+//! * [`unwrap`] panics with a generic message
+//! * [`unwrap_or`] returns the provided default value
+//! * [`unwrap_or_default`] returns the default value of the type `T`
+//!   (which must implement the [`Default`] trait)
+//! * [`unwrap_or_else`] evaluates a provided function
+//!
+//! [`Default`]: crate::default::Default
+//! [`expect`]: Option::expect
+//! [`unwrap`]: Option::unwrap
+//! [`unwrap_or`]: Option::unwrap_or
+//! [`unwrap_or_default`]: Option::unwrap_or_default
+//! [`unwrap_or_else`]: Option::unwrap_or_else
+//!
+//! ## Transforming contained values
+//!
+//! * [`map`] transforms [`Some<T>`] to [`Some<U>`] using the provided
+//!   function
+//! * [`map_or`] transforms [`Some<T>`] to a value of `U` using the
+//!   provided function, or transforms [`None`] to a provided default value
+//!   of `U`
+//! * [`map_or_else`] transforms [`Some<T>`] to [`Some<U>`] using the
+//!   provided function, or transforms [`None`] to a value of `U` using
+//!   another provided function
+//! * [`ok_or`] transforms [`Some(v)`] to [`Ok(v)`], and [`None`] to
+//!   [`Err(err)`] using the provided default `err` value
+//! * [`ok_or_else`] transforms [`Some(v)`] to [`Ok(v)`], and [`None`] to
+//!   a value of [`Err<E>`] using the provided function
+//!
+//! [`Err(err)`]: Err
+//! [`map`]: Option::map
+//! [`map_or`]: Option::map_or
+//! [`map_or_else`]: Option::map_or_else
+//! [`Ok(v)`]: Ok
+//! [`ok_or`]: Option::ok_or
+//! [`ok_or_else`]: Option::ok_or_else
+//! [`Some(v)`]: Some
+//!
+//! ## Boolean operators
+//!
+//! These methods treat the [`Option`] as a boolean value, where [`Some`]
+//! acts like [`true`] and [`None`] acts like [`false`]. There are two
+//! categories of these methods: ones that take an [`Option`] as input, and
+//! ones that take a function as input (to be lazily evaluated).
+//!
+//! The [`and`], [`or`], and [`xor`] methods take another [`Option`] as
+//! input, and produce an [`Option`] as output. Only the [`and`] method can
+//! produce an [`Option<U>`] value having a different inner type `U` than
+//! [`Option<T>`].
+//!
+//! | method  | self      | input     | output    |
+//! |---------|-----------|-----------|-----------|
+//! | [`and`] | N/A       | `None`    | `None`    |
+//! | [`and`] | `Some(x)` | `Some(y)` | `Some(y)` |
+//! | [`or`]  | `None`    | `None`    | `None`    |
+//! | [`or`]  | `None`    | `Some(y)` | `Some(y)` |
```

Comment:
This is also a don't care and not `None`

Reply:
Do you mean the input=`Some(y)` lines for `or`? I think that's not right for it to be a "don't care". For an actual two-valued logic that would be a "don't care", but there are multiple "true" values. `Some(x).or(Some(y))` is `Some(x)`, not `Some(y)` because of short-circuiting (though both are evaluated because `or` takes an expression and not a function).

Reply:
You're right, I'm stupid 😅